vtex-sites/nextjs.store
# Feat: Adjust Newsletter to match new designs

## What's the purpose of this pull request?

Updates the `Newsletter` component implementation according to the prototype in Figma. (Some updates were made while developing it)

## How does it work?

- Updates `Newsletter` component style (adds `card` and `lite` variations) [Preview](https://nextjs-store-storybook-git-feat-fs-4941-adjust-c017a7-faststore.vercel.app/?path=/docs/organisms-newsletter--default-story)

|Default - Deskstop| Default - Mobile|
|-|-|
<img width="1069" alt="image" src="https://user-images.githubusercontent.com/3356699/195652709-68140e07-26b3-4c03-ab39-cb0bd8f21cf3.png">|<img width="360" alt="image" src="https://user-images.githubusercontent.com/3356699/195652859-715ee8a1-31d9-4f6e-a89b-ccf475837c58.png">|

- Uses `Toast` to display Newsletter subscription feedback
- Adds `Newsletter` to Homepage and PDP
- Adds `Privacy Policy` addendum message to `Newsletter`
- Adds Toast Tokens documentation ([Preview](https://nextjs-store-storybook-git-feat-fs-4941-adjust-c017a7-faststore.vercel.app/?path=/docs/atoms-toast--default-story))

## How to test it?

- Check if `Newsletter` is being displayed on [Homepage](https://sfj-6dc3661--nextjs.preview.vtex.app/)

| Homepage - Mobile| Homepage - Desktop|
|-|-|
|<img width="333" alt="image" src="https://user-images.githubusercontent.com/3356699/195654608-51a5d94b-8570-4faf-a9df-e616f04b7a64.png">|<img width="1439" alt="image" src="https://user-images.githubusercontent.com/3356699/195654553-9706ed3e-36a6-4881-910c-016ced6198e5.png">|
 
- Check if `Toast` is being displayed correctly when subscribing 

|Error| Success|
|-|-|
|<img width="326" alt="image" src="https://user-images.githubusercontent.com/3356699/195654882-80a6ec8c-931f-4c2e-8084-759ba1508dcd.png">|<img width="328" alt="image" src="https://user-images.githubusercontent.com/3356699/195655142-44d8aad6-4308-4026-8b3a-890af6b0507c.png">|


- Check if `BannerNewsletter` is being displayed on [PDP page ](https://sfj-6dc3661--nextjs.preview.vtex.app/4k-philips-monitor-99988213/p)

|PDP - Mobile| PDP - Desktop|
|-|-|
|<img width="329" alt="image" src="https://user-images.githubusercontent.com/3356699/195770970-b97e281b-9306-409c-b4f1-75900c4617b3.png">|<img width="772" alt="image" src="https://user-images.githubusercontent.com/3356699/195771010-ebbd588b-e035-4889-9300-2a798e306ffa.png">|

## References

[FS-494](https://vtex-dev.atlassian.net/browse/FS-494?atlOrigin=eyJpIjoiODQyNmQ1Mjg1MDUwNGVlM2I4ZWRhNTc3YWMxY2I2MTEiLCJwIjoiaiJ9)

## Checklist

<em>You may erase this after checking them all :wink:</em>

**PR Title and Commit Messages**
  - Available prefixes: `feat`, `fix`, `chore`, `docs`, `style`, `refactor`, `perf`, and `test`

**PR Description**
- [x] Added a label according to the PR goal - `Breaking change`, `Features`, `Bug fixes`, `Chore`, `Documentation`, `Style changes`, `Refactoring`, `Performance`, and `Test`
- [x] Added the component, hook, or path name in-between backticks (\`\`) - *if applicable, e.g., `ComponentName` component, `useWindowDimensions` hook*

**Dependencies**

**Documentation**
